tests/testthat/test-tmb.R

```diff
@@ -376,6 +376,21 @@ test_that("h_mmrm_tmb_data catches case with multiple time points per subject ea
   )
 })
 
+
+test_that("h_mmrm_tmb_data has no side effect of overwrite the weights in global env", {
+  weights <- c("this is global weights")
+  formula <- FEV1 ~ RACE + SEX + ARMCD * AVISIT + us(AVISIT | USUBJID)
+  formula_parts <- h_mmrm_tmb_formula_parts(formula)
+  res <- h_mmrm_tmb_data(
+    formula_parts,
+    data = fev_data,
+    weights = rep_len(1, nrow(fev_data)),
+    reml = TRUE,
+    accept_singular = TRUE
+  )
+  expect_identical(weights, "this is global weights")
+})
+
```

Comment:
if sticking to current solution, please add test that checks that
1) `weights` in incoming data is not allowed
2) unless `weights` in incoming data is the same as specified as weights